javascript/ql/src/Statements/UselessConditional.ql

```diff
@@ -134,10 +134,13 @@ where guard.getTest() = op.asExpr() and
       // otherwise we just report that `op` always evaluates to `cv`
       else (
         sel = op.asExpr().stripParens() and
-        if sel instanceof VarAccess then
-          msg = "Variable '" + sel.(VarAccess).getVariable().getName() + "' always evaluates to " + cv + " here."
-        else
-          msg = "This expression always evaluates to " + cv + "."
+        exists (string description |
+          description = describeExpression(sel) |
+          if sel instanceof VarAccess then
+            msg = capitalize(description) + " always evaluates to " + cv + " here."
+          else
+            msg = "This " + description + " always evaluates to " + cv + "."
```

Comment:
In general we could get messages like:
```
This property 'f' always evaluates to true.
```
In general I think "named" expressions should belong in the former category.
```
Property 'f' always evaluates to true here.
```
We could add a boolean flag to `describeExpression` indicating whether it is named.

Reply:
The problem is the two kinds of messages. How about we conflate them to the latter, with variables being described as "read of variable 'v'":

Examples:
```
This comparison always evaluate to true.
This expression always evaluate to true.
This read of variable 'v' always evaluate to true.
This read of property 'p' always evaluate to true.
This call to 'f' always evaluate to true.
This property read always evaluate to true.
This call always evaluate to true.
```

Reply:
I would prefer `This use of` instead of `This read of`.

Reply:
Done + some additional features. I am not sure I like messages with the callee name though: they are long, and the callee name is not a link.